### DISCUSSION TITLE
how to disable default MediaResource

### DISCUSSION THREAD

**Discussion by mialyGit** [awcodes/filament-curator#180] [Q&A]
how to disable **default** MediaResource (all default media routes) by filament-curator and use the my custom media resource like this :

use Awcodes\Curator\Resources\MediaResource;

class CustomMediaResource extends MediaResource
{
    // ... custom methods and properties
}

**Comment by awcodes** [comment]
In a service provider you can tell the container to use your extensions instead of the plugins.

```php
public function register()
{
    $this->app->bind(MediaResource::class, fn() => new CustomMediaResource());
}
```

**Comment by mialyGit** [comment]
![Media problem](https://github.com/awcodes/filament-curator/assets/50203282/3d37a4a4-81f6-46a7-a9c0-3e2c6c9ba5bf)
Thanks  @awcodes  but The problem I have 2 media in the menu
I want to remove the one which created  by filament-curator 
How to fix it ?

**Reply by awcodes** [reply]
Ah, yea, that's registered at the plugin level. I'll have to build in a way for you to remove them.

**Comment by awcodes** [comment]
v2.5.10 should get you what you need. You just need to call `disableResourceRegistration()` on when defining global settings for Curator.

public function register()
{
    Curator::disableResourceRegistration();
    $this->app->bind(MediaResource::class, fn() => new YourNotAsCoolMediaResource());
    $this->app->bind(MediaResource\EditMedia::class, fn() => new YourNotAsCoolEditMedia());
    $this->app->bind(MediaResource\CreateMedia::class, fn() => new YourNotAsCoolCreateMedia());
    $this->app->bind(MediaResource\ListMedia::class, fn() => new YourNotAsCoolListMedia());
}
